CMakeLists.txt

```diff
@@ -1,20 +1,37 @@
 cmake_minimum_required(VERSION 3.15...3.26)
 project(${SKBUILD_PROJECT_NAME} LANGUAGES NONE)
 
+find_package(
+  Python
+  COMPONENTS Interpreter
+  REQUIRED)
 
-set(idc_index_release_version "0.3.2")
-set(idc_index_data_url "https://github.com/ImagingDataCommons/idc-index/releases/download/${idc_index_release_version}/idc_index.csv.zip")
-set(idc_index_data_sha256 "70ec9f915686a27bee3098163b8695c69c8696c05bfb7bd76943a24024cdeeb9")
+if(NOT DEFINED ENV{GCP_PROJECT})
+  message(FATAL_ERROR "GCP_PROJECT env. variable is not set")
+endif()
+
+option(IDC_INDEX_DATA_GENERATE_CSV_ARCHIVE "Generate idc_index.csv.zip file" ON)
+option(IDC_INDEX_DATA_GENERATE_PARQUET "Generate idc_index.parquet file" OFF)
 
-#
-# Download and install index
-#
 set(download_dir "${PROJECT_BINARY_DIR}")
-include(FetchContent)
-FetchContent_Populate(s5cmd
-  URL ${idc_index_data_url}
-  URL_HASH SHA256=${idc_index_data_sha256}
-  DOWNLOAD_DIR ${download_dir}
-  DOWNLOAD_NO_EXTRACT TRUE
-  )
-install(FILES "${download_dir}/idc_index.csv.zip" DESTINATION "idc_index_data")
+
+add_custom_command(
+  OUTPUT
+    $<$<BOOL:IDC_INDEX_DATA_GENERATE_CSV_ARCHIVE>:${download_dir}/idc_index.csv.zip>
```

Comment:
@jcfr if we want to build this for growth from the start, we cannot hard-code the file names. The generator code will now produce CSV/Parquet for each of the queries in the sql folder, but any changes to the SQL queries file names or addition of new queries will break the code at the moment.

It would make more sense to use wildcards to package all CSV/Parquet files depending on what is configured with the flags. Vamsi is going to give it a try to replace with the wildcard, unless you have other thoughts.

Reply:
> Vamsi is going to give it a try to replace with the wildcard

Using wildcard as such will not work as the target check the timestamp of the output (aka the filename) to decide if it should "re-build" or not.

Once this is integrated, I suggest you create an issue describing the type, name and size of index files, and how these related to queries for generating them (as well as which domain they relates) (e.g microscopy, ...)

Let's also not that with working CI and CD pipeline refactoring will be straightforward.